### PR TITLE
fix: log to stderr and other minor logging issues

### PR DESCRIPTION
**Pull request by dpc**

### CODE REVIEW

File: fedimint-logging/src/lib.rs [hunk 1]
```diff
@@ -52,7 +53,9 @@ impl TracingSetup {
     pub fn init(&self) -> anyhow::Result<()> {
         let filter_layer =
             EnvFilter::try_from_default_env().unwrap_or_else(|_| EnvFilter::new("info"));
-        let fmt_layer = tracing_subscriber::fmt::layer().with_filter(filter_layer);
+        let fmt_layer = tracing_subscriber::fmt::layer()
+            .with_writer(io::stderr)
```
Comment (garyray-k) [inline fedimint-logging/src/lib.rs:57]: Is there a reason tracing is written to stderr and not stdout?

Comment (dpc) [reply fedimint-logging/src/lib.rs:57]: Logging to stdout can interfere with normal output of each program.

Comment (Maan2003) [reply fedimint-logging/src/lib.rs:57]: :eyes: fmt layer uses stdout by default?

**Review by Maan2003 — APPROVED** [review]

**Review by elsirion — APPROVED** [review]

File: fedimint-logging/src/lib.rs (hunk 1 above)
Comment (dpc) [reply fedimint-logging/src/lib.rs:57]: @Maan2003 It seems like it. I filled https://github.com/tokio-rs/tracing/issues/2492 to complain / ask about it.

**Pull request merged**